src/middleware.js

```diff
@@ -0,0 +1,39 @@
+import { NextResponse } from "next/server";
+import { session } from "./lib/server-session";
+import * as jose from "jose";
+
+export async function middleware(request) {
+  //parsing the URL
+  const url = new URL(request.url);
+  const path = url.pathname;
+  //verify token on dashboard URL
+  if (path === "/dashboard") {
```

Comment:
This wont work if the `path` is /dashboard/create

you have to check if the path [__starts with__](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith) `/dashboard`. Not only equal to